test/w3c/domparsing.js

```diff
@@ -0,0 +1,74 @@
+"use strict";
+// http://www.w3.org/TR/DOM-Parsing/
+// (which is referenced by http://html.spec.whatwg.org/ )
+
+const jsdom = require("../../");
+
+exports["Setting outerHTML on the documentElement is not allowed"] = function (t) {
+  let doc = jsdom.jsdom("<html/>");
+
+  t.throws(function () {
+    doc.documentElement.outerHTML = "<html><head><title></title></head><body></body></html>";
+  }, doc.defaultView.DOMException);
```

Comment:
We should test the exception type. We could add some sort of test util somewhere and use it so that this becomes `t.throws(function () { ... }, domExceptionPredicate("NoModificationAllowedError"))` or similar.